src/libs/actions/Policy.js

```diff
@@ -512,9 +512,12 @@ function clearAvatarErrors(policyID) {
 function updateGeneralSettings(policyID, name, currency) {
     const optimisticData = [
         {
-            onyxMethod: Onyx.METHOD.MERGE,
+            // We use SET because it's faster than merge and avoids a race condition when setting the currency and navigating the user to the Bank account page in confirmCurrencyChangeAndHideModal
+            onyxMethod: Onyx.METHOD.SET,
```

Comment:
Coming from https://github.com/Expensify/App/issues/37254#issuecomment-1965696407

Because we're using `Onyx.set`, we should have spread the old pendingFields.